server/lib/multipartRequest.js

```diff
@@ -1,7 +1,12 @@
 import Promise from 'bluebird';
 import { ArgumentError } from 'auth0-extension-tools';
 
-export default function(client, entity, opts = {}, perPage = 50, concurrency = 5) {
+
+export default function(client, entity, opts = {}, fetchOptions = {} ) {
+  const perPage = fetchOptions.perPage || 50;
+  const concurrency = fetchOptions.concurrency || 5;
+  const limit = fetchOptions.limit || null;
```

Comment:
`Limit` is a bit confusing. 
If we pass `perPage: 100` - it will fetch 100 connections. Limit is not used for fetching. 
It is used for calculating `pageCount`, but even does not set limit on the number of already fetched connections.

Reply:
Maybe instead of limit it makes more sense to pass a param called 'pageCount'? Because we are using limit only for the purpose to get pageCount.

Reply:
I agree that it is a bit confusing! Yes, it is just used for calculating page count, which determines how many calls we make to API2. I guess we could change it to be pageCountLimit and use that number directly but I suppose this approach is slightly more flexible.

Reply:
It's flexible, but feels like bug when you realise its behaviour. I was super surprised to see `perPage = 100` and `limit = 1` still returns 100 connections.